app/src/main/java/com/nutomic/syncthingandroid/util/ConfigXml.java

```diff
@@ -342,14 +356,34 @@ private void saveChanges() {
         Log.i(TAG, "Writing updated config file");
         File mConfigTempFile = Constants.getConfigTempFile(mContext);
         try {
+            // Write XML header.
+            FileOutputStream fileOutputStream = new FileOutputStream(mConfigTempFile);
+            fileOutputStream.write("<?xml version=\"1.0\" encoding=\"UTF-8\"?>".getBytes("UTF-8"));
+
+            // Prepare Object-to-XML transform.
             TransformerFactory transformerFactory = TransformerFactory.newInstance();
             Transformer transformer = transformerFactory.newTransformer();
-            DOMSource domSource = new DOMSource(mConfig);
-            StreamResult streamResult = new StreamResult(mConfigTempFile);
-            transformer.transform(domSource, streamResult);
+            transformer.setOutputProperty(OutputKeys.METHOD, "xml");
+            transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-16");
+            transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+            transformer.setOutputProperty(OutputKeys.INDENT, "no");
+
```

Comment:
Is there a reason this is needed? It's nice to have the config readable.

Reply:
I preferred to have the config readable as I often use adb pull config.xml during debug sessions.

Reply:
But this says 'no' to indent, which I assume means not to indent it, which makes it a single long line?

Reply:
When I made a load and save test the indent was preserved like it was before.

Reply:
And if you remove this line all together, does it still have indentation? 

Reason I am asking is because for the next reader it will be completely not obvious why this line is here when it's not respected and the original intent is not clear either.

If it still does what it needs without this line, let's just remove it.

Reply:
I did this months ago and tested as long until I found a way where the transformation is still working when the encoding was changed. I'll check this when I have to update the ConfigXml next time in the fork and report back. Maybe I can do the work once instead of twice like this to find the optimal way.